crates/bevy_ecs/src/batching.rs

```diff
@@ -0,0 +1,108 @@
+//! Types for controlling batching behavior during parallel processing.
+
+use std::ops::Range;
+
+/// Dictates how a parallel operation chunks up large quanitities
+/// during iteration.
+///
+/// A parallel query will chunk up large tables and archetypes into
+/// chunks of at most a certain batch size. Similarly, a parallel event
+/// reader will chunk up the remaining events.
+///
+/// By default, this batch size is automatically determined by dividing
+/// the size of the largest matched archetype by the number
+/// of threads (rounded up). This attempts to minimize the overhead of scheduling
+/// tasks onto multiple threads, but assumes each entity has roughly the
+/// same amount of work to be done, which may not hold true in every
+/// workload.
+///
+/// See [`Query::par_iter`], [`EventReader::par_read`] for more information.
+///
+/// [`Query::par_iter`]: crate::system::Query::par_iter
+/// [`EventReader::par_read`]: crate::event::EventReader::par_read
+#[derive(Clone, Debug)]
+pub struct BatchingStrategy {
+    /// The upper and lower limits for how large a batch of entities.
```

Comment:
```suggestion
    /// The upper and lower limits for how large is a batch of entities.
```
Looks like something is missing here
at the same way, we can solve this by
```suggestion
    /// The upper and lower limits for a batch of entities.
```